app/src/main/java/com/example/bigowlapp/repository/NotificationRepository.java

```diff
@@ -24,17 +19,11 @@ public NotificationRepository() {
         super("notifications");
     }
 
-    // TODO: Remove or Modify when dependency injection implemented
-    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
-    public NotificationRepository(FirebaseFirestore mFirebaseFirestore, CollectionReference collectionReference) {
-        super(mFirebaseFirestore, collectionReference);
-    }
-
     public MutableLiveData<List<SupervisionRequest>> getListOfSupervisionRequestByAttribute(String attribute, String attrValue,
                                                                                             Class<? extends SupervisionRequest> tClass) {
         MutableLiveData<List<SupervisionRequest>> listOfTData = new MutableLiveData<>();
         collectionReference.whereEqualTo(attribute, attrValue)
-                .whereEqualTo("type", Constants.SUPERVISION_TYPE)
+                .whereEqualTo("type", Notification.Type.SUPERVISION_REQUEST)
```

Comment:
Consider using a magic literal and refactor the code; string "type" should be kept in a variable.